src/app/toolbar/saved-chats-panel/saved-chats-panel.component.ts

```diff
@@ -0,0 +1,122 @@
+import {Component, HostListener} from '@angular/core';
+import {MatDialogRef} from "@angular/material/dialog";
+import {MessageService} from "../../services/message.service";
+import {HighlightService} from "../../services/highlight.service";
+import {ChatCompletionRequestMessage} from "openai/dist/api";
+import {PageEvent} from "@angular/material/paginator";
+
+@Component({
+  selector: 'app-saved-chats-panel',
+  templateUrl: './saved-chats-panel.component.html',
+  styleUrls: ['./saved-chats-panel.component.css']
+})
+export class SavedChatsPanelComponent {
+  constructor(private dialogRef: MatDialogRef<SavedChatsPanelComponent>, private messageService: MessageService, private highlightService: HighlightService) {
+  }
+
+  steps: number;
+  messageHistory: Chat[] = [];
+  messageHistoryPaginated: Chat[] = [];
+  chatSelected: boolean = false;
+  aboutToDelete: boolean = false;
+  chatName: string = "";
+  currentLoadedChatName: string = "";
+  containerHeight: string = '400px';
+
+
+  ngOnInit(): void {
+    if (localStorage.getItem('chatHistory') === null) {
+      this.messageHistory = [];
+    } else {
+      this.messageHistory = JSON.parse(localStorage.getItem('chatHistory'));
+      console.log(JSON.parse(localStorage.getItem('chatHistory')));
+
+      this.containerHeight = this.calculateAccordionHeight();
+    }
+    if (this.messageService.currentChatName.length > 0) {
```

Comment:
I get an error sometimes here on the currentChatName as it is undefined sometimes. Might need to initalize it with an empty string or something.
![image](https://user-images.githubusercontent.com/1709120/232235249-d14335c3-f4fc-47d8-8fa6-5679fbfcdf33.png)